kernel/tasking/Syscalls.cpp

```diff
@@ -78,6 +78,11 @@ Result sys_process_cancel(int pid)
 
     Task *task = task_by_id(pid);
 
+    if (!task->user)
+    {
+        return ERR_ACCESS_DENIED;
+    }
+
     if (task == nullptr)
```

Comment:
You put the user check before the null check, enjoy your null deref/kernel panic :p

Reply:
big oof